Remove derivation checks for regression tests

BTom-GH opened the pull request:
Remove derivation check by adding -d=0 in the Makefile for all tests (see #609). Also, fix the parsing in the Python file (it detected some "rule" in the warning part as an additional rule, which generated bogus "The rules do not match").

There are still warnings because the message derivation check warnings have changed. If we want these warnings to disappear, I believe that the case-studies-regression files should be updated.

BTom-GH committed:
Parse better and remove derivation check in regression test (but there is still an error while running with slow parameter)

BTom-GH committed:
temp fix for running -s

BTom-GH committed:
Merge branch 'tamarin-prover:develop' into fix-remove-derivation-checks-regression-tests

BTom-GH committed:
remove ignoring tests not working on my computer.

jdreier commented:
Thanks!

We should wait for the other PRs impacting the regression tests to be merged, and then update the outdated targets within this PR before merging.

@rsasse We were unable to find regression tests for the derivation checks, can you help us here?

rsasse commented:
> Thanks!
> 
> We should wait for the other PRs impacting the regression tests to be merged, and then update the outdated targets within this PR before merging.
> 
> @rsasse We were unable to find regression tests for the derivation checks, can you help us here?

I had thought there were dedicated regression tests for the derivation checks, however after checking https://github.com/tamarin-prover/tamarin-prover/pull/541 I realized that there are not. This made sense, as derivation checks applied to ALL theories in the regressions and thus had been covered.

Turning derivation checks off for all tests in the makefile is thus not ideal. Rather, at least some subset (where some warnings actually show up) should still have derivation checks turned on.

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier commented:
> Turning derivation checks off for all tests in the makefile is thus not ideal. Rather, at least some subset (where some warnings actually show up) should still have derivation checks turned on.

I think adding some dedicated tests for the derivation checks makes more sense. They can be fast (thus we can avoid the non-determinism issue), and we can test more effectively.

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier commented:
We also need to update the regression targets before merging this PR.

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

BTom-GH committed:
update targets with github's artifact for fast tests

BTom-GH committed:
First few derivation checks tests

BTom-GH commented:
Also fix issues like #592

BTom-GH committed:
Some more tests

BTom-GH committed:
Better comments on derivation tests files.

BTom-GH committed:
Update makefile target and case-studies-regression

BTom-GH committed:
Update non-problematic slow tests

BTom-GH committed:
Adding derivation check tests to slow test targets

BTom-GH committed:
update case-studies-regression with missing targets

BTom-GH committed:
Better parsing of analyzed files. (Still need to be cleaned from debug commented prints)

BTom-GH committed:
Update issue193 target

BTom-GH committed:
Update regressionTests parsing.
For rules :
Comparison is quote-independent and more specific : it checks name, attributes and body separately.
Still need to clean debug prints

BTom-GH committed:
update slow targets

BTom-GH committed:
clean debug prints

BTom-GH commented:
The targets should all be updated now.
However i think that accountability tests were removed, should i also remove them from the makefile ?
Moreover, regression/trace/issue193 isn't in the fast cases, should we add it ?

rkunnema commented:
Hi! The accountability tests should stay in the regression tests. Or is this just about derivation checks. Can you give more details? (I'll be afk for two weeks now, maybe @kevinmorio can check for me.)

BTom-GH commented:
I was indeed talking about the regression tests, because it seems like they have been removed at some point in #738 ([this](https://github.com/tamarin-prover/tamarin-prover/pull/738/commits/588e1f2c7359b033a184c910f7fd138a46bda8c2) commit i believe ?)
But if it was an error then i'll simply add them back, no worries.
For regression/trace/issue193, it's only in the slow regression tests for now, is it normal or should i also add it (back ?) to the fast tests ?

jdreier commented:
I would add the accountability tests back, I think they should be part of the tests. I would also add regression/trace/issue193 to the fast tests.

BTom-GH committed:
Re-add accountability to regression tests & add issue193 to fast tests

BTom-GH committed:
Forgot to remove commented unused code

rkunnema commented:
Ah, that's these specific accountability cases – I thought all of them (sorry, I am writing from vacation, little time to check commits.). Removing these specifically is fine then.

BTom-GH committed:
Forgot to update the makefile

jdreier commented:
@rkunnema it's only about adding the target files back for these accountability files. They are part of the slow checks currently, and we did not modify this.

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier committed:
merge and new regression targets

jdreier committed:
Updated regression targets

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier committed:
Merge

jdreier committed:
update regression targets

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests

jdreier committed:
fixed Makefile

jdreier committed:
update target

jdreier committed:
Merge branch 'develop' into fix-remove-derivation-checks-regression-tests
